src/common/components/repositories-home/index.js

```diff
@@ -89,10 +89,11 @@ class RepositoriesHome extends Component {
 
   render() {
     const { snaps } = this.props;
+    const hasSnaps = (snaps.snaps && Object.keys(snaps.snaps).length > 0);
     // show spinner until we know if user has any enabled repos
-    return (snaps.success || snaps.error)
-        ? this.renderRepositoriesList()
-        : this.renderSpinner();
+    return (snaps.isFetching && !hasSnaps)
+      ? this.renderSpinner()
+      : this.renderRepositoriesList();
```

Comment:
@earnubs: It was something I considered, but as @bartaz suggested we already have state in the store that can tell us when we're polling and when we're fetching for the initial status. In fact, we could probably write a selector to do this for us.